src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -471,7 +473,10 @@ def _compile_hermetic(self, jvm_options, ctx, classes_dir, jar_file,
       native_image_snapshots = []
       # TODO: Lean on distribution for the bin/java appending here
       image_specific_argv =  ['.jdk/bin/java'] + jvm_options + [
-        '-cp', zinc_relpath,
+        # put all the scala deps on the classpath so that if they change and we're using nailgun through
+        # zinc's nailMain method we will invalidate the nailgun if the scala paths change. The path itself
+        # is used for the fingerprint not the content on the jars.
```

Comment:
Which jar do you mean when you say "the jar", nailgun? We need the nailgun jar on the classpath before zinc so that the correct nailgun jar gets picked up. ZInc's internal jar is not the version specified in BUILD.tools

Reply:
No: I mean "the scala deps" that this comment is referring to.